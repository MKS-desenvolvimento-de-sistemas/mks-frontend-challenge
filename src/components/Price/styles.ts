import styled from "styled-components";

interface IPriceProps {
  type: "product" | "cart";
}

export const Price = styled.div<IPriceProps>`
  padding: ${(props) => (props.type === "product" ? "5px 10px" : "8px 17px")};

  background-color: var(--secondary-color-100);
  border-radius: 5px;

  display: flex;
  justify-content: center;
  align-items: center;

  font-weight: 700;
  font-size: 15px;
  line-height: 15px;
  text-align: center;
  color: var(--primary-color-100);

  @media (min-width: 760px) {
    padding: ${(props) => (props.type === "product" ? "4px 7px" : "0")};

    background-color: ${(props) => props.type === "cart" && "transparent"};
    color: ${(props) => props.type === "cart" && "var(--secondary-color-300)"};
  }
`;
