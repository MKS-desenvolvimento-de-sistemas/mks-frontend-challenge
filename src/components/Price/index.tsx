import * as Styles from "./styles";

interface IPriceProps {
  price: number;
  type: "product" | "cart";
}

const Price = ({ price, type = "product" }: IPriceProps) => {
  return <Styles.Price type={type}>{`R$${price}`}</Styles.Price>;
};

export default Price;
