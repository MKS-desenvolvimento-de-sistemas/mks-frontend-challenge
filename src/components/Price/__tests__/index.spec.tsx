import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import Price from "../";
import renderer from "react-test-renderer";

describe("Price Component", () => {
  it("should render the price with type product correctly", () => {
    render(<Price price={200} type="product" />);

    const price = screen.getByTestId("price");

    expect(price).toHaveTextContent("R$200");
    expect(price).toHaveStyle({
      padding: "5px 10px" || "4px 7px",
    });
  });

  it("should render the price with type cart correctly", () => {
    render(<Price price={200} type="cart" />);

    const price = screen.getByTestId("price");

    expect(price).toHaveTextContent("R$200");
    expect(price).toHaveStyle({
      padding: "8px 17px" || "0",
    });
  });

  it("should match the snapshot - type product", () => {
    const tree = renderer.create(<Price price={200} type="product" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should match the snapshot - type cart", () => {
    const tree = renderer.create(<Price price={200} type="cart" />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
