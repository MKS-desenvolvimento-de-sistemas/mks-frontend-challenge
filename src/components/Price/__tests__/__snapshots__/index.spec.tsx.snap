// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price Component should match the snapshot - type cart 1`] = `
<div
  className="sc-bczRLJ bjpjYw"
  data-testid="price"
  type="cart"
>
  R$200
</div>
`;

exports[`Price Component should match the snapshot - type product 1`] = `
<div
  className="sc-bczRLJ kwDUOH"
  data-testid="price"
  type="product"
>
  R$200
</div>
`;
